### PR TITLE
docs: passwordless auth docs improvements.

### PR DESCRIPTION
**Pull request by thorwebdev**
## What kind of change does this PR introduce?

* add top level `Passwordless login` nav
* Add Email OTP guide

### CODE REVIEW

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    // set this to false if you do not want the user to be automatically signed up
+    shouldCreateUser: false,
+  },
+})
+```
+
+</TabPanel>
+<TabPanel id="dart" label="Dart">
+
+When your user signs in, call [signIn()](/docs/reference/dart/auth-signinwithotp) with their email address:
```
Comment (charislam) [inline apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:49]:
```suggestion
When your user signs in, call [signInWithOtp()](/docs/reference/dart/auth-signinwithotp) with their email address:
```

I think based on the Dart reference?

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+```dart
+Future<void> signInWithEmailOtp() async {
+  final AuthResponse res = await supabase.auth.signinwithotp(email: 'example@email.com');
```
Comment (charislam) [inline apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:53]:
```suggestion
  final AuthResponse res = await supabase.auth.signInWithOtp(email: 'example@email.com');
```

I think based on the Dart reference?

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    "user": null,
+    "session": null
+  },
+  "error": null
+}
+```
+
+## Verify OTP to create session
+
+Provide an input field for the user to key in the one-time code. To verify the code and complete the user's sign in, call [verifyOtp()](/docs/reference/javascript/auth-verifyotp) with their email address, the code, and `type: "email"`:
```
Comment (charislam) [inline apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:87]: For consistency, should this also have the Dart and Kotlin sample code?

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  "token_type": "bearer",
+  "expires_in": 3600,
+  "refresh_token": "LSp8LglPPvf0DxGMSj-vaQ",
+  "user": {...}
+}
+```
+
+## Resources
+
+- [Supabase Account - Free Plan OK](https://supabase.com)
```
Comment (charislam) [inline apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:114]: Is this the intended link?

**Review by charislam — APPROVED** [review]

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts [hunk 5]
```diff
@@ -497,7 +497,15 @@ export const auth = {
...
       items: [
         { name: 'Email Login', url: '/guides/auth/auth-email' },
-        { name: 'Magic Link Login', url: '/guides/auth/auth-magic-link' },
+        {
+          name: 'Passwordless Login',
+          url: '/guides/auth/passwordless-login',
+          items: [
+            { name: 'Email Magic Link', url: '/guides/auth/passwordless-login/auth-magic-link' },
+            { name: 'Email OTP', url: '/guides/auth/passwordless-login/auth-email-otp' },
+            { name: 'Phone OTP', url: '/guides/auth/phone-login' },
```
[the remark this answers is not in the document]
Comment (thorwebdev) [reply apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:506]: My thinking behind this is that folks have different paths of looking for things. So in order to meet them where they would look, I think it's worthwhile to duplicate some of these items. Wdyt?

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx (hunk 4 above)
Comment (thorwebdev) [reply apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:114]: I actually don't know where these are coming from. I *think* their purpose is to show that you can use this feature with a free Supabase account?

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (charislam) [reply apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:506]: yeah I'm torn on this one, I agree with the intent but find the jumping menu sections behavior a bit unexpected. I'm fine with leaving for now, we can try it out!

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx (hunk 4 above)
Comment (charislam) [reply apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:116]: I think if we want to say that we should just add a sentence somewhere to say it 😄 and take out this link as it's confusing

File: apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx (hunk 3 above)
Comment (thorwebdev) [reply apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:89]: Ah sorry, I missed this. Will leave it to the community to contribute 😅

Comment (thorwebdev) [reply apps/docs/pages/guides/auth/passwordless-login/auth-email-otp.mdx:89]: Ah sorry, I missed this. Will leave it to the community to contribute 😅

**Pull request merged**